tests/countries/test_sierra_leone.py

```diff
@@ -0,0 +1,180 @@
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from unittest import TestCase
+
+from holidays.countries import SierraLeone, SL, SLE
+from tests.common import CommonCountryTests
+
+
+class TestSierraLeone(CommonCountryTests, TestCase):
+    @classmethod
+    def setUpClass(cls):
+        years = range(1962, 2050)
+        super().setUpClass(SierraLeone, years=years, years_non_observed=years)
+
+    def test_country_aliases(self):
+        self.assertAliases(SierraLeone, SL, SLE)
+
+    def test_no_holidays(self):
+        self.assertNoHolidays(SierraLeone(years=1960))
+        self.assertNoHolidays(SierraLeone(years=1961))
+
+    def test_new_years_day(self):
+        self.assertHolidayName("New Year's Day", (f"{year}-01-01" for year in range(1962, 2050)))
+
+    def test_armed_forces_day(self):
+        name = "Armed Forces Day"
+        self.assertHolidayName(name, (f"{year}-02-18" for year in range(2002, 2050)))
+        self.assertNoHolidayName(name, range(1962, 2002))
+
+    def test_international_womens_day(self):
+        name = "International Women's Day"
+        self.assertHolidayName(name, (f"{year}-03-08" for year in range(2018, 2050)))
+        self.assertNoHolidayName(name, range(1962, 2018))
+
+    def test_good_friday(self):
+        name = "Good Friday"
+        self.assertHolidayName(
+            name, "2022-04-15", "2023-04-07", "2024-03-29", "2025-04-18", "2026-04-03"
+        )
+        self.assertHolidayName(name, range(1962, 2050))
+
+    def test_easter_monday(self):
+        name = "Easter Monday"
+        self.assertHolidayName(
+            name, "2022-04-18", "2023-04-10", "2024-04-01", "2025-04-21", "2026-04-06"
+        )
+        self.assertHolidayName(name, range(1962, 2050))
+
+    def test_independence_day(self):
+        name = "Independence Day"
+        self.assertHolidayName(name, (f"{year}-04-27" for year in range(1962, 2050)))
+        dt = (
+            "2008-04-28",
+            "2013-04-29",
+            "2019-04-29",
+            "2024-04-29",
+        )
+        self.assertHolidayName(f"{name} (observed)", dt)
+        self.assertNoNonObservedHoliday(dt)
+
+    def test_labor_day(self):
+        name = "International Worker's Day"
+        self.assertHolidayName(name, (f"{year}-05-01" for year in range(1962, 2050)))
+
+    def test_christmas_day(self):
+        name = "Christmas Day"
+        self.assertHolidayName(name, (f"{year}-12-25" for year in range(1962, 2050)))
+
+    def test_boxing_day(self):
+        name = "Boxing Day"
+        self.assertHolidayName(name, (f"{year}-12-26" for year in range(1962, 2050)))
+
+    def test_prophets_birthday(self):
+        name = "Prophet's Birthday (estimated)"
+        self.assertHolidayName(
+            name,
+            "2018-11-21",
+            "2019-11-10",
+            "2020-10-29",
+            "2021-10-18",
+            "2022-10-08",
+            "2023-09-27",
+            "2024-09-15",
+        )
```

Comment:
I recommend looking at Islamic holidays tests in other countries (for example, Indonesia or Philippines).